### ISSUE TITLE
@vitest/coverage-v8 is shown as a false positive

### ISSUE THREAD

**Issue by EricSmekens** [depcheck/depcheck#909]
### Bug Description

When you have a project, where you use `vitest` to run tests and to generate coverage, `@vitest/coverage-v8` is used as default coverage analyzer.

https://vitest.dev/guide/coverage.html

```
Unused devDependencies
* @vitest/coverage-v8
```

### Code snippets

**Where the dependency is used:**

This is a part of vitest.config.ts. The `provider: 'v8'` is optional, as v8 is set as default within vitest.

You'll get the same false positive, when you leave out the provider line.

```typescript
export default defineConfig({
  test: {
    onConsoleLog() {
      return false;
    },
    coverage: {
      reporter: ['text', 'lcov', 'cobertura'],
      include: ['src/**'],
      exclude: ['src/schemas/**'],
      provider: 'v8'
    },
    reporters: ['junit', 'verbose'],
    outputFile: {
      junit: './test/results/junit.xml',
    },
    globals: true,```
```

That coverage is actually 'used', is hard to detect, as it is only run when vitest is ran, with the `run --coverage` cli option.
```
    "test:coverage": "vitest run --coverage",
```
Where the dependency is listed in package.json:

```
 "devDependencies": {
    "@types/minimist": "^1.2.2",
    "@types/node": "^18.15.11",
    "@typescript-eslint/eslint-plugin": "^5.58.0",
    "@typescript-eslint/parser": "^5.58.0",
    "@vitest/coverage-v8": "^2.1.1",
    "dotenv": "^16.0.3",
    "eslint": "^8.38.0",
    "minimist": "^1.2.8",
    "msw": "^2.4.11",
    "nodemon": "^3.1.4",
    "orval": "^7.1.0",
    "rimraf": "^5.0.0",
    "ts-node": "^10.9.1",
    "typescript": "^5.0.4",
    "vite-plugin-swc-transform": "^0.4.0",
    "vitest": "^2.1.1"
  },
```

### Versions

- `node -v`: 20.18.0
- `npm -v`: 10.8.2
- `depcheck --version`: 1.4.7

### Extra info

Any extra info you want to include. Thank you for the bug report!

BONUS POINTS: If you wish to help us debug further, the `depcheck --json` option will show more information. Check [here in the README](https://github.com/depcheck/depcheck#example) for documentation on json output.

**Comment by hpandelo** [comment]
No updates? I just faced it today
